File: app/src/main/java/net/emojiparty/android/jishotomo/data/models/PrimaryOnlyEntry.java
package net.emojiparty.android.jishotomo.data.models;

import androidx.room.ColumnInfo;

public class PrimaryOnlyEntry {
  public int id;
  @ColumnInfo(name = "primary_kanji")
  public String primaryKanji;
  @ColumnInfo(name = "primary_reading")
  public String primaryReading;

  public String getKanjiOrReading() {
    return hasKanji() ? primaryKanji : primaryReading;
  }

  public String getReading() {
    return hasKanji() ? primaryReading : null;
  }

  public boolean hasKanji() {
    return primaryKanji != null;
    }
}

File: app/src/main/java/net/emojiparty/android/jishotomo/data/models/PrimaryOnlyEntry.kt
package net.emojiparty.android.jishotomo.data.models

import androidx.room.ColumnInfo

// TODO revisit inheritance of this class (SearchResultEntry and CrossReferencedEntry)
data class PrimaryOnlyEntry(
  var id: Int,
  @ColumnInfo(name = "primary_kanji") var primaryKanji: String? = null,
  @ColumnInfo(name = "primary_reading") var primaryReading: String
) {
  fun hasKanji(): Boolean = primaryKanji != null

  fun kanjiOrReading(): String? = if (hasKanji()) primaryKanji else primaryReading

  fun reading(): String? = if (hasKanji()) primaryReading else null

}
